### PR TITLE
Replace alertmanager flag with config file option

### PR DESCRIPTION
**Pull request by fabxc**
@juliusv

### CODE REVIEW

File: notification/notification.go [hunk 1]
```diff
@@ -74,8 +76,8 @@ type httpPoster interface {
 // NotificationHandler is responsible for dispatching alert notifications to an
 // alert manager service.
 type NotificationHandler struct {
-	// The URL of the alert manager to send notifications to.
-	alertmanagerURL string
+	// The URLs of the alert manager to send notifications to.
```
Comment (juliusv) [inline notification/notification.go]: s/alertmanager/alertmanagers/

Comment (fabxc) [inline notification/notification.go]: Done.

**Comment by juliusv** [comment]
:+1: If we're going this route, we'll also want to move the long-term storage URLs into the config later.

/cc @brian-brazil FYI

**Comment by brian-brazil** [comment]
Should we wait on the alertmanager redesign before making this change? No point in making more breaking changes than we have to.

**Comment by fabxc** [comment]
The earlier we make it the less hassle for user to upgrade later.
Let's be realistic here, a distributed Alertmanager is far away – we simply don't have the man-power. And with much later I mean after Prometheus 1.0, after which we cannot break anything.

**Comment by brian-brazil** [comment]
I doubt that this will be how we'll end up configuring the alertmanager with the distributed setup, we're likely going to need to support SD, all the auth options, relabelling etc.

> And with much later I mean after Prometheus 1.0, after which we cannot break anything.

We can still break things then, we just need to be more measured in how we do so. We can also point to parts of the system we still consider unstable even with 1.0.

**Comment by fabxc** [comment]
I'd rather have a distributed Alertmanager being self contained. A client should know about a number of servers and sync with them about coming and leaving peers as they. Relying on SD is not ideal.

**Comment by brian-brazil** [comment]
It's hard to predict right now exactly what sort of operational setup we'll end up with, but I'm fairly certain no matter what we advise some users will want SD particularly for the more global ones like Consul.

**Pull request closed without merging**